src/lib/stitching/convection/v2/stitching.ts

```diff
@@ -90,5 +100,45 @@ export const consignmentStitchingEnvironment = (
           ),
       },
     },
+    Mutation: {
+      createConsignmentSubmission: {
+        resolve: async (_source, args, context, info) => {
+          const myCollectionArtworkID = args.input?.myCollectionArtworkID
+          let createSubmissionArgs = args
+
+          // when myCollectionArtworkID is specified, use artwork data to fill in submission
+          if (myCollectionArtworkID) {
+            if (context.artworkLoader) {
+              const artwork = await context.artworkLoader(myCollectionArtworkID)
+
+              if (!artwork) {
+                throw new Error("Artwork not found")
```

Comment:
Maybe this is easier to debug 🤷 

```suggestion
                throw new Error(`My Collection artwork (ID: ${myCollectionArtworkID}) not found.`)
```